defender/tests/Governance_Twitter_Bot.spec.js

```diff
@@ -0,0 +1,252 @@
+const mockTweet = jest.spyOn(TwitterApi.prototype.v1, 'tweet').mockImplementation(() => ({ id_str: '1' }));
+// reply will prepend a 1 to the tweetID that it was passed. '1' => '11' => '111'
+const mockReply = jest.spyOn(TwitterApi.prototype.v1, 'reply').mockImplementation((msg, id) => ({ id_str: `1${id}` }));
+
+// expect(person.sayMyName()).toBe("Hello");
+// expect(person.bla()).toBe("bla");
+
+const { handler } = require('../governance_twitter_bot/autotask-1/index');
+
+describe('check autotask', () => {
+  beforeEach(() => {
+    mockContract.state.mockClear();
+    mockContract.quorumVotes.mockClear();
+    mockAxios.mockClear();
+    mockTweet.mockClear();
+    mockReply.mockClear();
+
+    // define default values
+    mockContract.state.mockReturnValue(0); // Default to pending state
+    mockContract.comp.mockReturnValue(`0x${'1'.repeat(40)}`); // 0x1111...
+    // need 100 Tokens to reach quorum
+    mockContract.quorumVotes.mockReturnValue(ethers.BigNumber.from(`100${'0'.repeat(mockDecimals)}`));
+
+    // one block per second
+    mockProvider.getBlock.mockClear();
+    mockProvider.getBlock.mockImplementation((block) => {
+      switch (block) {
+        case 'latest':
+          return {
+            number: 1000, // Block 1000
+            timestamp: 1000,
+          };
+        default:
+          return {
+            number: 0, // Block 0
+            timestamp: 0,
+          };
+      }
+    });
+
+    mockAxios.get.mockImplementation((id) => {
+      switch (id) {
+        case 'https://api.compound.finance/api/v2/governance/proposals?proposal_ids[]=1':
+          return {
+            data: { proposals: [{ title: 'Prop1' }] },
+          };
+        case 'https://api.compound.finance/api/v2/governance/proposals?proposal_ids[]=2':
+          return {
+            data: { proposals: [{ title: 'Prop2' }] },
+          };
+        default:
+          return null;
+      }
+    });
+
+    mockContract.proposals.mockReset();
+    mockContract.proposals.mockReturnValue(exampleProposal);
+  });
+
+  afterEach(() => {
+  });
+
+  it('does nothing if no proposals have been created', async () => {
+    mockContract.initialProposalId = jest.fn().mockResolvedValueOnce(ethers.BigNumber.from(0));
+    mockContract.proposalCount = jest.fn().mockResolvedValueOnce(ethers.BigNumber.from(0));
+
+    await handler({ secrets });
+
+    // no proposals should be looked up
+    expect(mockContract.state).toBeCalledTimes(0);
+    expect(mockContract.proposals).toBeCalledTimes(0);
+    expect(mockAxios).toBeCalledTimes(0);
+  });
+
+  it('does nothing for proposals that are not active', async () => {
+    mockContract.initialProposalId = jest.fn().mockResolvedValueOnce(ethers.BigNumber.from(0));
+    mockContract.proposalCount = jest.fn().mockResolvedValueOnce(ethers.BigNumber.from(6));
+
+    mockContract.state = jest.fn()
+      .mockResolvedValueOnce(0) // Pending
+      // .mockResolvedValueOnce(1) // Active (will test active state in next test)
```

Comment:
```suggestion

    // Active case is omitted
    mockContract.state = jest.fn()
      .mockResolvedValueOnce(0) // Pending
```

Reply:
Suggested to get rid of the commented out code on line 155 and added a comment on line 153 before mocking the contract state

Reply:
Done, addressed in f75166284eb9fd2891dd653545cd2be50558c400